src/Sylius/Component/Payment/Resolver/CompositeMethodsResolver.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Payment\Resolver;
+
+use Sylius\Component\Payment\Model\PaymentInterface;
+use Sylius\Component\Registry\PrioritizedServiceRegistryInterface;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class CompositeMethodsResolver implements MethodsResolverInterface
+{
+    /**
+     * @var PrioritizedServiceRegistryInterface
+     */
+    private $resolversRegistry;
+
+    /**
+     * @param PrioritizedServiceRegistryInterface $resolversRegistry
+     */
+    public function __construct(PrioritizedServiceRegistryInterface $resolversRegistry)
+    {
+        $this->resolversRegistry = $resolversRegistry;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSupportedMethods(PaymentInterface $payment)
+    {
+        foreach ($this->resolversRegistry->all() as $resolver) {
+            if ($resolver->supports($payment)) {
+                return $resolver->getSupportedMethods($payment);
+            }
+        }
+
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(PaymentInterface $payment)
+    {
+        return true;
```

Comment:
See [this](https://github.com/Sylius/Sylius/blob/c1655e50b6c7133c7151999cd7233a6e24cc8205/src/Sylius/Component/Shipping/Resolver/CompositeMethodsResolver.php#L54) for a proper solution.